test/gtest/ucp/ucp_test.h

```diff
@@ -81,10 +81,11 @@ class ucp_test_base : public ucs::test_base {
                const ucp_worker_params_t& worker_params,
                const ucp_test_base* test_owner);
 
-        ~entity();
+        virtual ~entity();
```

Comment:
Making these `virtual` can negatively impact overall CI running time. Please measure.

An alternative would be to replace inheritance (`test_ucp_configure::entity`) with composition / friend class.

Reply:
from what I measured it seems perf impact is negligible.